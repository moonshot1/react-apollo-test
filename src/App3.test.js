import React from 'react';

import { MockedProvider } from 'react-apollo/lib/test-utils';
import TestRenderer from 'react-test-renderer';

import { Query } from 'react-apollo';
import gql from 'graphql-tag';


// The component AND the query need to be exported
const QUERY = gql`
  query HeroFriends($episode: Episode!) {
    hero(episode: $episode) {
      name
    }
  }
`;

const HeroAndFriends = ({ episode }) => (
  <Query query={QUERY} variables={{ episode }}>
    {({ data, error, loading }) => {
      if (error) return 'Oops!';
      if (loading) return 'Patience young grasshopper...';
      console.log("data obj", data);
      return (
        <div>
          <h1>{data.hero.name}</h1>

        </div>
      );
    }}
  </Query>
)

const mocks = [
  {
    request: {
      query: QUERY,
      variables: {
        episode: 'NEWHOPE',
      },
    },
    result: {
      hero: { name: 'Luke' }
    },
  },
];

it('renders without error', () => {
  TestRenderer.create(
    <MockedProvider mocks={mocks} addTypename={false}>
      <HeroAndFriends episode="NEWHOPE" />
    </MockedProvider>
  );
});