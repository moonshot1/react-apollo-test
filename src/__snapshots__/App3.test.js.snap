// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without error 1`] = `
<ApolloProvider
  client={
    ApolloClient {
      "cache": InMemoryCache {
        "addTypename": false,
        "config": Object {
          "addTypename": false,
          "dataIdFromObject": [Function],
          "fragmentMatcher": HeuristicFragmentMatcher {},
          "storeFactory": [Function],
        },
        "data": ObjectCache {
          "data": Object {},
        },
        "optimistic": Array [],
        "silenceBroadcast": false,
        "typenameDocumentCache": WeakMap {},
        "watches": Array [],
      },
      "defaultOptions": Object {},
      "disableNetworkFetches": false,
      "link": ApolloLink {
        "request": [Function],
      },
      "mutate": [Function],
      "query": [Function],
      "queryDeduplication": true,
      "reFetchObservableQueries": [Function],
      "resetStore": [Function],
      "resetStoreCallbacks": Array [],
      "ssrMode": false,
      "store": DataStore {
        "cache": InMemoryCache {
          "addTypename": false,
          "config": Object {
            "addTypename": false,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "storeFactory": [Function],
          },
          "data": ObjectCache {
            "data": Object {},
          },
          "optimistic": Array [],
          "silenceBroadcast": false,
          "typenameDocumentCache": WeakMap {},
          "watches": Array [],
        },
      },
      "version": "2.3.5",
      "watchQuery": [Function],
    }
  }
>
  <App3 />
</ApolloProvider>
`;
